src/client-functions/client-function-factory.js

```diff
@@ -46,46 +48,65 @@ export default class ClientFunctionFactory {
 
         var factory = this;
 
-        clientFn.bindTestRun = function bindTestRun (t) {
-            // NOTE: we can't use strict `t instanceof TestController`
-            // check due to module circular reference
-            if (!t || !(t.testRun instanceof TestRun))
-                throw new APIError('bindTestRun', MESSAGE.invalidClientFunctionTestRunBinding);
+        clientFn.with = function (options) {
```

Comment:
As far as I remember in that case Babel desugaring causes problems with callsite. But I'll give it a try.